configure

```diff
@@ -228,10 +225,15 @@ while true; do
       cudnn_lib_path="${CUDNN_INSTALL_PATH}/lib/libcudnn.dylib"
       cudnn_alt_lib_path="${CUDNN_INSTALL_PATH}/libcudnn.dylib"
     fi
+    # Resolve to the SONAME of the symlink.  Use readlink without -f
+    # to resolve exactly once to the SONAME.  E.g, libcudnn.so ->
+    # libcudnn.so.4.
+    # If the path is not a symlink, readlink will exit with an error code, so
+    # in that case, we return the path itself.
     if [ -f "$cudnn_lib_path" ]; then
-      REALVAL=`readlink -f ${cudnn_lib_path}`
+      REALVAL=`readlink ${cudnn_lib_path} || echo "${cudnn_lib_path}"`
```

Comment:
Are we looking for the full path? echo seems to be not doing the same thing as readlink -f.